nuxeo-management-rest-api/src/test/java/org/nuxeo/rest/management/TestElasticsearchObject.java

```diff
@@ -0,0 +1,156 @@
+        // Creates a new docs without indexing them
+        createDocuments();
+
+        // Wait for an eventual indexation
+        TransactionHelper.startTransaction();
+        waitForIndexing();
+        assertEquals(0, ess.query(new NxQueryBuilder(coreSession).nxql(GET_DOCUMENTS_QUERY)).totalSize());
+
+        // Starts the ES indexing
+        String path = String.format("site/management/elasticsearch/%s/reindex", coreSession.getRepositoryName());
+        try (CloseableClientResponse response = httpClientRule.post(path, null)) {
+            assertEquals(SC_NO_CONTENT, response.getStatus());
+            waitForIndexing();
+            assertEquals(2 + initialDocCount,
+                    ess.query(new NxQueryBuilder(coreSession).nxql(GET_DOCUMENTS_QUERY)).totalSize());
+        }
+    }
+
+    @Test
+    public void shouldRunFlush() {
+        try (CloseableClientResponse response = httpClientRule.post("site/management/elasticsearch/flush", null)) {
+            assertEquals(SC_NO_CONTENT, response.getStatus());
+        }
+    }
+
+    @Test
+    public void shouldRunFlushOnRepository() {
+        String path = String.format("site/management/elasticsearch/%s/flush", coreSession.getRepositoryName());
+        try (CloseableClientResponse response = httpClientRule.post(path, null)) {
+            assertEquals(SC_NO_CONTENT, response.getStatus());
+        }
+    }
+
+    @Test
+    public void shouldRunOptimize() {
+        try (CloseableClientResponse response = httpClientRule.post("site/management/elasticsearch/optimize", null)) {
+            assertEquals(SC_NO_CONTENT, response.getStatus());
+        }
+    }
+
+    @Test
+    public void shouldRunOptimizeOnRepository() {
+        String path = String.format("site/management/elasticsearch/%s/optimize", coreSession.getRepositoryName());
+        try (CloseableClientResponse response = httpClientRule.post(path, null)) {
+            assertEquals(SC_NO_CONTENT, response.getStatus());
+        }
+    }
+
+    @Test
+    public void shouldFailRunningIndexationWhenRepositoryNotExist() {
+        // Launch the ES indexing
+        String path = String.format("site/management/elasticsearch/%s", "unExistingRepository");
+        try (CloseableClientResponse response = httpClientRule.post(path, null)) {
+            assertEquals(SC_NOT_FOUND, response.getStatus());
+        }
+    }
+
+    protected void createDocuments() {
+        Stream.of("my-file-1", "my-file-2").forEach((fileName) -> {
+            DocumentModel doc = coreSession.createDocumentModel("/default-domain/workspaces/", fileName, "File");
+            doc.setPropertyValue("dc:title", String.format("Title of %s", fileName));
+            doc.putContextData(ElasticSearchConstants.DISABLE_AUTO_INDEXING, Boolean.TRUE);
+            coreSession.createDocument(doc);
+        });
+
+        coreSession.save();
+        TransactionHelper.commitOrRollbackTransaction();
+    }
+
+    protected void waitForIndexing() throws Exception {
```

Comment:
`TransactionalFeature#nextTransaction` should wait for the indexing.

Reply:
I just had a look, and the waiter for elasticsearch indexing is in `RepositoryElasticsearchFeature` (not deployed here) and only available in 11.1.
For the work manager, we have `WorkManagerFeature` which waits for it since 10.3.

Reply:
I think that this only the case if the `RepositoryElasticSearchFeature` is used.
And I'm not sure that this was backported to 10.10, see https://jira.nuxeo.com/browse/NXP-27330.

Reply:
replace the call to `waitForIndexing` by `TransactionalFeature#nextTransaction` the `esa.refresh();` is needed (i comment this line and the test fails) what i propose is tu keep the `waitForIndexing` but i will replace it by :

`
    protected void waitForIndexing() {
        txFeature.nextTransaction();
        esa.refresh();
    }
`
wdyt